torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1336,7 +1395,11 @@ void ProcessGroupNCCL::workCleanupLoop() {
         ++it;
       }
     }
-
+    // process a request to dump the trace
+    if (dumpPipe.shouldDump()) {
+      std::thread dump_thread(&ProcessGroupNCCL::dumpDebuggingInfo, this);
```

Comment:
to avoid duplicate write, can we also set `writeDebugInfo_` to false?

Reply:
It is setup so that writing multiple times to the pipe can cause a dump to be produced again later. That way it doesn't interfere with a later attempt to dump on timeout, or prevent a dump from being requested later in training.